components/gitpod-cli/pkg/utils/trackEvent.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package utils
+
+import (
+	"context"
+	"time"
+
+	gitpod "github.com/gitpod-io/gitpod/gitpod-cli/pkg/gitpod"
+	"github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor"
+	serverapi "github.com/gitpod-io/gitpod/gitpod-protocol"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// Rebuild
+	RebuildErrorCode_DockerBuildFailed     = "rebuild_docker_build_failed"
+	RebuildErrorCode_DockerErr             = "rebuild_docker_err"
+	RebuildErrorCode_DockerfileCannotRead  = "rebuild_dockerfile_cannot_read"
+	RebuildErrorCode_DockerfileCannotWirte = "rebuild_dockerfile_cannot_write"
+	RebuildErrorCode_DockerfileEmpty       = "rebuild_dockerfile_empty"
+	RebuildErrorCode_DockerfileNotFound    = "rebuild_dockerfile_not_found"
+	RebuildErrorCode_DockerNotFound        = "rebuild_docker_not_found"
+	RebuildErrorCode_DockerRunFailed       = "rebuild_docker_run_failed"
+	RebuildErrorCode_MalformedGitpodYaml   = "rebuild_malformed_gitpod_yaml"
+	RebuildErrorCode_MissingGitpodYaml     = "rebuild_missing_gitpod_yaml"
+	RebuildErrorCode_NoCustomImage         = "rebuild_no_custom_image"
+)
+
+type TrackCommandUsageParams struct {
+	Command                    string  `json:"command,omitempty"`
+	DurationMs                 int64   `json:"durationMs,omitempty"`
+	ErrorCode                  string  `json:"errorCode,omitempty"`
+	WorkspaceId                string  `json:"workspaceId,omitempty"`
+	InstanceId                 string  `json:"instanceId,omitempty"`
+	Timestamp                  int64   `json:"timestamp,omitempty"`
+	DockerBuildDurationSeconds float64 `json:"dockerBuildDurationSeconds,omitempty"`
```

Comment:
Let’s call it imageBuildDurationSeconds. We are likely go away from docker next week to build the image.